src/hotspot/src/share/vm/prims/unsafe.cpp

```diff
@@ -167,11 +167,18 @@ jint Unsafe_invocation_key_to_method_slot(jint key) {
 
 #define GET_FIELD(obj, offset, type_name, v) \
   oop p = JNIHandles::resolve(obj); \
-  type_name v = *(type_name*)index_oop_from_field_offset_long(p, offset)
+  JavaThread* t = JavaThread::current(); \
```

Comment:
Not a correctness issue, but this may slow down GET_FIELD on all platforms: current() can be expensive.

Reply:
index_oop_from_field_offset() can be expensive, so adding this code probably won't make a difference.